leshan-server-cf/src/main/java/org/eclipse/leshan/server/californium/impl/CaliforniumLwM2mRequestSender.java

```diff
@@ -44,14 +46,15 @@
     // A map which contains all pending CoAP requests
     // This is mainly used to cancel request and avoid retransmission on de-registration
     private final ConcurrentNavigableMap<String/* registrationId#requestId */, Request /* pending coap Request */> pendingRequests = new ConcurrentSkipListMap<>();
+    private final ConcurrentLinkedQueue<ResponseListener> responseListeners = new ConcurrentLinkedQueue<ResponseListener>();
```

Comment:
Also, I think you should use the abstract interface instead of the concrete class when declaring the field, i.e.
```
private final Queue<ResponseListener> responseListeners = new ConcurrentLinkedQueue<>();
```

unless you are referring to some non-Queue-defined methods ...